### PR TITLE
feat(ci): Rust CircleCI Checks

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Moves rust checks to CircleCI with Slack alerting on failure for `main` branch runs.

### CODE REVIEW

**Comment by refcell** [comment]
Converting to a draft to work on this a bit more - need to figure out the best way to perform releases while also hitting slack on failure.

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ orbs:
   go: circleci/go@1.8.0
   node: circleci/node@5.2.0
   slack: circleci/slack@4.10.1
+  rust: circleci/rust@1.6.1
```
Comment (clabby) [inline .circleci/config.yml:7]: Is this the rust version, or just the package version? 1.6.1 super old compiler

**Review by clabby — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (refcell) [reply .circleci/config.yml:7]: Just the orb version - latest publish was december 2023, so should be recent enough.
See: https://circleci.com/developer/orbs/orb/circleci/rust

**Review by geoknee — APPROVED** [review]
LGTM! Enjoying the justfile updates

**Pull request merged**